### PR TITLE
Move LLVM submodule updates back to native.rs

### PR DESCRIPTION
**Pull request by jyn514**
Time to find more bugs!

The first commit is a straight revert of https://github.com/rust-lang/rust/pull/85647, the second is a fix for https://rust-lang.zulipchat.com/#narrow/stream/182449-t-compiler.2Fhelp/topic/x.2Epy.20always.20updates.20LLVM.20submodule/near/240113320 and https://github.com/rust-lang/rust/pull/82653#issuecomment-846755631. I haven't been able to replicate https://github.com/rust-lang/rust/pull/82653#issuecomment-849013698.

### CODE REVIEW

File: src/bootstrap/native.rs [hunk 1]
```diff
@@ -91,6 +91,89 @@ pub fn prebuilt_llvm_config(
...
+            .current_dir(&build.config.src),
+    );
+    let hash = recorded
+        .split_whitespace()
+        .nth(2)
+        .unwrap_or_else(|| panic!("unexpected output `{}`", recorded));
+
+    // update_submodule
+    if let Some(llvm_hash) = checked_out {
+        if hash == llvm_hash {
```
Comment (LingMan) [inline src/bootstrap/native.rs:136]: This check only gets reached if the `build.config.fast_submodules` above is true. Maybe move `recorded`, `hash`, and this check inside the `if` and get rid of this Option boilerplate?

Comment (jyn514) [reply src/bootstrap/native.rs]: Good idea, thanks!

**Comment by Mark-Simulacrum** [comment]
Seems OK -- r=me, but I would prefer to land this when you have some cycles to either revert it or fix bugs that may pop up. Obviously, you know better than I when that is :)

**Comment by jyn514** [comment]
Thanks :) I should have time this Friday afternoon

**Comment by durin42** [comment]
FWIW, this appears to have the same behavior mentioned in #85647 that it unconditionally updates LLVM even if `[build] submodules = false` is specified. I'm not sure if that's intentional, but it'd be nice to have a way to have LLVM not get updated when doing builds. Should I file a bug?

**Comment by jyn514** [comment]
> FWIW, this appears to have the same behavior mentioned in #85647 that it unconditionally updates LLVM even if `[build] submodules = false` is specified. I'm not sure if that's intentional, but it'd be nice to have a way to have LLVM not get updated when doing builds. Should I file a bug?

Yes, please do. Thanks!

**Comment by durin42** [comment]
Filed #86954, thanks!

**Pull request merged**